go-controller/pkg/factory/factory.go

```diff
@@ -1313,6 +1323,10 @@ func (wf *WatchFactory) addHandler(objType reflect.Type, namespace string, sel l
 			return true, nil
 		})
 		if err != nil {
+			// handler is not going to be added, restore previous value if needed
+			if hadZeroHandlers {
+				atomic.StoreUint32(&intInf.hasHandlers, hasNoHandler)
```

Comment:
i just realized on handler removal i need to set it back to 0 if there are no handlers left on the internal informer. I'll do a follow up after I see how CI goes here

Reply:
done